### ISSUE TITLE
Allow Multiple VPN-Connections 

### ISSUE THREAD

**Issue by mathishuettl** [hwdsl2/setup-ipsec-vpn#216]
Hello,

how can I allow multiple VPN-Connections?
I created two Users with this tutorial: https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/manage-users.md

When UserA is connected UserB can't connect.
When UserA disconnects UserB can connect, but UserA can't connect.

**Comment by hwdsl2** [comment]
@mathishuettl Hello! As mentioned in the "important notes" section [1] of the README: Due to an IPsec/L2TP limitation, if you wish to connect multiple devices simultaneously from behind the same NAT (e.g. home router), you must use only IPsec/XAuth mode [2] instead of IPsec/L2TP.

[1] https://github.com/hwdsl2/setup-ipsec-vpn#important-notes
[2] https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients-xauth.md

**Comment by Kyslik** [comment]
@hwdsl2 I've done just that and it still kicks me out.

I set up multiple users using the guide. I use OS X and iPhone to test this and DO droplet; any hints how to diagnose this?

(yes I've set up both devices using IPsec/XAuth readme)

---

I test it in a way that I ping google from my mac and when I hit connect on iPhone all ICMP packets are timing out `Request timeout for icmp_seq 256`...

Logs from `auth.log` on the server (redacted usernames, redacted ips), at `Feb  3 21:17:32` I hit connect on iPhone:

```
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: responding to Main Mode from unknown peer 98.765.432.1 on port 500
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: STATE_MAIN_R1: sent MR1, expecting MI2
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: STATE_MAIN_R2: sent MR2, expecting MI3
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: ignoring informational payload IPSEC_INITIAL_CONTACT, msgid=00000000, length=28
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: | ISAKMP Notification Payload
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: |   00 00 00 1c  00 00 00 01  01 10 60 02
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: Peer ID is ID_IPV4_ADDR: '192.168.5.62'
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: STATE_MAIN_R3: sent MR3, ISAKMP SA established {auth=PRESHARED_KEY cipher=aes_256 integ=sha group=MODP1024}
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: Sending Username/Password request (MAIN_R3->XAUTH_R0)
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: password file authentication method requested to authenticate user 'redacted-mac'
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: password file (/etc/ipsec.d/passwd) open.
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: success user(redacted-mac:xauth-psk)
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: #15: completed for user 'redacted-mac' with status SUCCESSS
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: User redacted-mac: Authentication Successful
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: XAUTH: xauth_inR1(STF_OK)
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: STATE_MAIN_R3: sent MR3, ISAKMP SA established {auth=PRESHARED_KEY cipher=aes_256 integ=sha group=MODP1024}
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: modecfg_inR0(STF_OK)
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: STATE_MODE_CFG_R1: ModeCfg Set sent, expecting Ack {auth=PRESHARED_KEY cipher=aes_256 integ=sha group=MODP1024}
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #15: the peer proposed: 0.0.0.0/0:0/0 -> 192.168.43.10/32:0/0
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16: responding to Quick Mode proposal {msgid:6e8e3e9c}
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16:     us: 0.0.0.0/0===123.456.78.9[MS+XS+S=C]
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16:   them: 98.765.432.1[+MC+XC+S=C]===192.168.43.10/32
Feb  3 21:17:15 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16: STATE_QUICK_R1: sent QR1, inbound IPsec SA installed, expecting QI2 tunnel mode {ESP/NAT=>0x0a180861 <0x5f0844d2 xfrm=AES_CBC_256-HMAC_SHA1_96 NATOA=none NATD=98.765.432.1:4500 DPD=active username=redacted-mac}
Feb  3 21:17:16 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16: STATE_QUICK_R1: retransmission; will wait 0.5 seconds for response
Feb  3 21:17:16 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #16: STATE_QUICK_R2: IPsec SA established tunnel mode {ESP/NAT=>0x0a180861 <0x5f0844d2 xfrm=AES_CBC_256-HMAC_SHA1_96 NATOA=none NATD=98.765.432.1:4500 DPD=active username=redacted-mac}
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: responding to Main Mode from unknown peer 98.765.432.1 on port 6
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: STATE_MAIN_R1: sent MR1, expecting MI2
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: STATE_MAIN_R2: sent MR2, expecting MI3
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: ignoring informational payload IPSEC_INITIAL_CONTACT, msgid=00000000, length=28
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: | ISAKMP Notification Payload
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: |   00 00 00 1c  00 00 00 01  01 10 60 02
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: Peer ID is ID_IPV4_ADDR: '192.168.5.52'
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: STATE_MAIN_R3: sent MR3, ISAKMP SA established {auth=PRESHARED_KEY cipher=aes_256 integ=sha2_256 group=MODP2048}
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: Sending Username/Password request (MAIN_R3->XAUTH_R0)
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: password file authentication method requested to authenticate user 'redacted-iph'
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: password file (/etc/ipsec.d/passwd) open.
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: success user(redacted-iph:xauth-psk)
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: #17: completed for user 'redacted-iph' with status SUCCESSS
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: User redacted-iph: Authentication Successful
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: XAUTH: xauth_inR1(STF_OK)
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: STATE_MAIN_R3: sent MR3, ISAKMP SA established {auth=PRESHARED_KEY cipher=aes_256 integ=sha2_256 group=MODP2048}
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: modecfg_inR0(STF_OK)
Feb  3 21:17:32 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: STATE_MODE_CFG_R1: ModeCfg Set sent, expecting Ack {auth=PRESHARED_KEY cipher=aes_256 integ=sha2_256 group=MODP2048}
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #17: the peer proposed: 0.0.0.0/0:0/0 -> 192.168.43.10/32:0/0
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18: responding to Quick Mode proposal {msgid:b5c9c9d1}
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18:     us: 0.0.0.0/0===123.456.78.9[MS+XS+S=C]
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18:   them: 98.765.432.1[+MC+XC+S=C]===192.168.43.10/32
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18: keeping refhim=0 during rekey
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18: STATE_QUICK_R1: sent QR1, inbound IPsec SA installed, expecting QI2 tunnel mode {ESP/NAT=>0x020e23b1 <0x47b30139 xfrm=AES_CBC_256-HMAC_SHA1_96 NATOA=none NATD=98.765.432.1:1029 DPD=active username=redacted-iph}
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18: STATE_QUICK_R1: retransmission; will wait 0.5 seconds for response
Feb  3 21:17:33 debian-s-1vcpu-1gb-ams3-01-vpn pluto[6666]: "xauth-psk"[8] 98.765.432.1 #18: STATE_QUICK_R2: IPsec SA established tunnel mode {ESP/NAT=>0x020e23b1 <0x47b30139 xfrm=AES_CBC_256-HMAC_SHA1_96 NATOA=none NATD=98.765.432.1:1029 DPD=active username=redacted-iph}
Feb  3 21:18:21 debian-s-1vcpu-1gb-ams3-01-vpn sshd[6862]: Connection closed by 5.101.60.43 port 58657 [preauth]
```
